storage/database.go

```diff
@@ -356,17 +551,61 @@ func (d *db) namespaceFor(namespace ts.ID) (databaseNamespace, error) {
 	return n, nil
 }
 
-func (d *db) getOwnedNamespaces() []databaseNamespace {
-	d.RLock()
+func (d *db) ownedNamespacesWithLock() []databaseNamespace {
 	namespaces := make([]databaseNamespace, 0, len(d.namespaces))
 	for _, n := range d.namespaces {
 		namespaces = append(namespaces, n)
 	}
-	d.RUnlock()
 	return namespaces
 }
 
+func (d *db) GetOwnedNamespaces() []databaseNamespace {
+	d.RLock()
+	defer d.RUnlock()
+	return d.ownedNamespacesWithLock()
+}
+
 func (d *db) nextIndex() uint64 {
 	created := atomic.AddUint64(&d.created, 1)
 	return created - 1
 }
+
+type tsIDs []ts.ID
+
+func (t tsIDs) String() (string, error) {
+	if len(t) == 0 {
+		return "[]", nil
+	}
+	var buf bytes.Buffer
+	for idx, id := range t {
+		if idx != 0 {
+			if _, err := buf.WriteString(", "); err != nil {
+				return "", err
+			}
+		}
+		if _, err := buf.WriteString(id.String()); err != nil {
+			return "", err
+		}
+	}
+	return buf.String(), nil
+}
+
+type metadatas []namespace.Metadata
+
+func (m metadatas) String() (string, error) {
+	if len(m) == 0 {
+		return "[]", nil
+	}
+	var buf bytes.Buffer
```

Comment:
Same here regarding prefix and postfix with `[` and `]`.

Reply:
sure thing

Reply:
done